src/domain/swaps/contracts/decoders/set-pre-signature-decoder.helper.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { AbiDecoder } from '@/domain/contracts/decoders/abi-decoder.helper';
+import { getFunctionSelector, isHex, parseAbi } from 'viem';
+
+export const abi = parseAbi([
+  'function setPreSignature(bytes calldata orderUid, bool signed)',
+]);
+
+@Injectable()
+export class SetPreSignatureDecoder extends AbiDecoder<typeof abi> {
+  private readonly setPreSignatureFunctionSelector: `0x${string}`;
+
+  constructor() {
+    super(abi);
+    this.setPreSignatureFunctionSelector = getFunctionSelector(abi[0]);
+  }
+
+  /**
+   * Gets the Order UID associated with the provided transaction data.
+   *
+   * @param data - the transaction data for the setPreSignature call
+   * @returns {`0x${string}`} the order UID or null if the data does not represent a setPreSignature transaction
+   */
+  getOrderUid(data: `0x${string}`): `0x${string}` | null {
+    if (!data.startsWith(this.setPreSignatureFunctionSelector)) return null;
+    const { functionName, args } = this.decodeFunctionData({ data });
+    if (functionName !== 'setPreSignature') return null;
+    if (!args || !args[0] || typeof args[0] !== 'string' || !isHex(args[0]))
```

Comment:
These checks are theoretically not needed on a type level.

Are you checking for "fake" signatures? If so, you don't need to check `args[0]` or the `typeof` as the arguments are always an array (if present) and `isHex` accepts an `unknown` type.

Reply:
Right! At some point I had a state where the type checks were not working but `decodeFunctionData` does return the correct types. Fixed in [9dca8df](https://github.com/safe-global/safe-client-gateway/pull/1239/commits/9dca8df8a3cde5031ea9fd75fc93e15fb4a30639)